### ISSUE TITLE
Give photography tips?

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#2158]
I have spot a user who uploads pictures of many interesting places, but unfortunately always holding their phone in portrait position, leading to non-optimal pictures:
https://commons.wikimedia.org/w/index.php?title=Special:ListFiles/Liverpoolpics

After 20 uploads, should we detect that and if 90% of the pictures are in portrait mode, and if yes show something like below?

> Did you know?
> Most pictures of landscapes and buildings look better if taken with your phone rotate on its side
> (Picture of someone taking a nice picture of a building, in landscape mode)

Other tip ideas welcome, please comment below :-)

Low priority

**Comment by misaochan** [comment]
My suggestion is slightly more vague in scope, but still relevant I think. How about we have a "tip" section in a loading screen (say, when user is loading Nearby) that has little tips like this? Not just for photography, but also for categorization, etc.

**Comment by nicolas-raoul** [comment]
Great idea! The Duolingo app does this, showing short facts in the loading screen during which a lesson's data is downloaded from the Internet. The user's uploads could influence what tips they see (for instance if their last picture had zero selected categories).

**Comment by vanshikaarora** [comment]
I would like to work upon this. I am a beginner at open source could you please, assign me some task to work upon this.

**Comment by neslihanturan** [comment]
Thanks @vanshikaarora for your interest, I saw you also commented on #2165, please make sure you claimed one task at a time. #2165 or this one is fine. 

I think @misaochan 's suggestion is great. Maybe this information can be supported with some images, since it is faster interpret images, than text. And during loading, we have limited time to communicate with user.

**Comment by vanshikaarora** [comment]
Can you please be specific regarding the point categorization

**Comment by neslihanturan** [comment]
I think there should be a set of tips, and different texts/images (I am willing to send some images after we decide our titles) related with those tips. We can randomly pick one and display different tips (this could be fun). 

- One about promoting landscape
- One about EXIF
- One about categorization
- One about statistics 

Actually I loved displaying something when our app is IDLE, we can also share some thank messages or statistics with user. It is a good way to communicate with user. And when they feel there are someone else behind of this app, they would be more willing to send logs and communicate with us.

@vanshikaarora apparently this issue needs some more discussion that everyone else joined, so starting working on this can require some time.

**Comment by vanshikaarora** [comment]
Yes sir the idea is cool but discussion part is reaaly important. I'll wait till the discusion goes on and then will start the work

**Comment by misaochan** [comment]
Images would be cool! However the main problem with having a different image for every tip is (1) unnecessarily increasing APK size, and (2) every time we want to add a new tip, we need to get an image for it.

How about just a few different images, each one for a certain type of tip? For instance photography tips could all have a camera image. And categorization tips could have... well I dunno what image we'd use for that, but you get the idea. ;)

> Actually I loved displaying something when our app is IDLE, we can also share some thank messages or statistics with user. It is a good way to communicate with user. And when they feel there are someone else behind of this app, they would be more willing to send logs and communicate with us.

I agree.

Randomizing it would be great, too!

**Comment by neslihanturan** [comment]
Agree about APK size, maybe we can use text and image for some of them, and only text for others? So images are not mandatory.

**Comment by neslihanturan** [comment]
And we could draw some creative and attractive images ^_^

**Comment by vanshikaarora** [comment]
> I think there should be a set of tips, and different texts/images (I am willing to send some images after we decide our titles) related with those tips. We can randomly pick one and display different tips (this could be fun).
> 
> * One about promoting landscape
> * One about EXIF
> * One about categorization
> * One about statistics
> 
> Actually I loved displaying something when our app is IDLE, we can also share some thank messages or statistics with user. It is a good way to communicate with user. And when they feel there are someone else behind of this app, they would be more willing to send logs and communicate with us.
> 
> @vanshikaarora apparently this issue needs some more discussion that everyone else joined, so starting working on this can require some time.

Hey @neslihanturan have you decided the title and got the images?

**Comment by nicolas-raoul** [comment]
The app now tips the user to enabled camera location, if their last 10 uploads all lack location data _(thanks @Ayan-10 for implementing!)_.

Doing the same for these sounds doable:
- 10 last uploads are all in portrait mode
- 10 last uploads are all with zero category

About displaying general tips when idle, maybe create a new issue?

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul @4D17Y4 i would like to work on this issue please assign me

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 Thanks!
There is already a warning if the last 10 (I think) uploaded pictures do not contain latitude/longitude, hopefully you can reuse that and build upon it.

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 Thanks! There is already a warning if the last 10 (I think) uploaded pictures do not contain latitude/longitude, hopefully you can reuse that and build upon it.

can you please tell me on which file this function is implemented I am unable to find it in the project

**Comment by Rishavgupta12345** [comment]
> @Rishavgupta12345 Thanks! There is already a warning if the last 10 (I think) uploaded pictures do not contain latitude/longitude, hopefully you can reuse that and build upon it.

yes it was implemented in #4735 but I uploaded 12 pictures without location and didn't get any warning pop up or anything
